### PR TITLE
Support GPUBuffer input type and ArrayBufferView

### PR DESCRIPTION
**Pull request by fujunwei**

### CODE REVIEW

**Comment by fujunwei** [comment]
@huningxin  PTAL, thank.

File: src/webnn_native/GraphBuilder.cpp [hunk 1]
```diff
@@ -217,6 +226,12 @@ namespace webnn_native {
         VALIDATE_FOR_OPERAND(new op::Unary(this, op::UnaryOpType::kNeg, input));
     }
 
+    // OperandBase* GraphBuilderBase::Pad(OperandBase* input,
+    //                                    uint32_t const* padding,
+    //                                    size_t padding_count,
+    //                                    PadOptions const* options) {
+    //     VALIDATE_FOR_OPERAND(new op::Pad(this, input, padding, padding_count, options));
+    // }
```
Comment (huningxin) [inline src/webnn_native/GraphBuilder.cpp:234]: should we remove the commented code?

File: src/webnn_native/dml/GraphDML.h [hunk 2]
```diff
@@ -47,6 +46,24 @@
...
+#    include <wrl/implements.h>
+
+#    include <Windows.h>
+#    include <d3d12.h>
+
+#    define DML_TARGET_VERSION_USE_LATEST 1
+#    include <DirectML.h>
+#    include "DirectMLX.h"
+
+#    include "dawn/native/dml/deps/src/dmldevice.h"
```
Comment (huningxin) [inline src/webnn_native/dml/GraphDML.h:62]: Where is this file hosted? Is it added into the DEPS?

File: src/webnn_native/Instance.cpp [hunk 3]
```diff
@@ -161,6 +161,23 @@ namespace webnn_native {
...
+#if defined(WEBNN_ENABLE_GPU_BUFFER)
+        WGPUDevice device = reinterpret_cast<WGPUDevice>(wnn_device->device);
+        if (mBackends.find(wnn::BackendType::DirectML) != mBackends.end()) {
+            return mBackends[wnn::BackendType::DirectML]->CreateContextWithGpuDevice(device);
+        } else if (mBackends.find(wnn::BackendType::OpenVINO) != mBackends.end()) {
+            return mBackends[wnn::BackendType::OpenVINO]->CreateContextWithGpuDevice(device);
+        } else if (mBackends.find(wnn::BackendType::OneDNN) != mBackends.end()) {
+            return mBackends[wnn::BackendType::OneDNN]->CreateContextWithGpuDevice(device);
+        } else if (mBackends.find(wnn::BackendType::MLAS) != mBackends.end()) {
+            return mBackends[wnn::BackendType::MLAS]->CreateContextWithGpuDevice(device);
```
Comment (huningxin) [inline src/webnn_native/Instance.cpp:174]: My question is whether we support GPU device for backends other than DirectML?

File: build_overrides/webnn_features.gni [hunk 4]
```diff
@@ -33,4 +33,5 @@ declare_args() {
   webnn_enable_null = true
 
   webnn_enable_wire = false
+  webnn_enable_gpu_buffer = false
```
Comment (huningxin) [inline build_overrides/webnn_features.gni:36]: What's the dependency of this one? Does it require enable dml?

Comment (huningxin) [reply build_overrides/webnn_features.gni:36]: I mean it would be good to document it in the readme.

**Review by huningxin — COMMENTED** [review]
Thanks for the PR @fujunwei , it LGTM in high level. Just left a few questions, please take a look.

File: src/webnn_native/dml/GraphDML.h (hunk 2 above)
Comment (fujunwei) [reply src/webnn_native/dml/GraphDML.h:62]: It hosts in Dawn repo seeing https://github.com/fujunwei/dawn/commits/webnn_webgpu_interop which need to be refactored.

File: src/webnn_native/Instance.cpp (hunk 3 above)
Comment (fujunwei) [reply src/webnn_native/Instance.cpp:174]: Yes, i think so, all backend will receive the GPU device as parameter.

File: build_overrides/webnn_features.gni (hunk 4 above)
Comment (fujunwei) [reply build_overrides/webnn_features.gni:36]: How about documenting it in building Chromium which will be used in there?

**Comment by fujunwei** [comment]
Merge it, thanks.

File: build_overrides/webnn_features.gni (hunk 4 above)
Comment (huningxin) [reply build_overrides/webnn_features.gni:36]: It sounds good.

**Pull request merged**